pkg/objstore/providers/gcs/config.go

```diff
@@ -26,6 +28,30 @@ func (cfg *Config) RegisterFlags(f *flag.FlagSet) {
 func (cfg *Config) RegisterFlagsWithPrefix(prefix string, f *flag.FlagSet) {
 	f.StringVar(&cfg.BucketName, prefix+"gcs.bucket-name", "", "GCS bucket name")
 	f.Var(&cfg.ServiceAccount, prefix+"gcs.service-account", cfg.GCSServiceAccountShortDescription())
+	cfg.HTTP.RegisterFlagsWithPrefix(prefix, f)
+}
+
+type HTTPConfig struct {
+	IdleConnTimeout       time.Duration `yaml:"idle_conn_timeout"`
+	ResponseHeaderTimeout time.Duration `yaml:"response_header_timeout"`
+	InsecureSkipVerify    bool          `yaml:"insecure_skip_verify"`
+	TLSHandshakeTimeout   time.Duration `yaml:"tls_handshake_timeout"`
+	ExpectContinueTimeout time.Duration `yaml:"expect_continue_timeout"`
+	MaxIdleConns          int           `yaml:"max_idle_conns"`
+	MaxIdleConnsPerHost   int           `yaml:"max_idle_conns_per_host"`
+	MaxConnsPerHost       int           `yaml:"max_conns_per_host"`
+}
```

Comment:
Probably better to mark them all as advanced:

```suggestion
type HTTPConfig struct {
	IdleConnTimeout       time.Duration `yaml:"idle_conn_timeout"` `category:"advanced"`
	ResponseHeaderTimeout time.Duration `yaml:"response_header_timeout"` `category:"advanced"`
	InsecureSkipVerify    bool          `yaml:"insecure_skip_verify"` `category:"advanced"`
	TLSHandshakeTimeout   time.Duration `yaml:"tls_handshake_timeout"` `category:"advanced"`
	ExpectContinueTimeout time.Duration `yaml:"expect_continue_timeout"` `category:"advanced"`
	MaxIdleConns          int           `yaml:"max_idle_conns"` `category:"advanced"`
	MaxIdleConnsPerHost   int           `yaml:"max_idle_conns_per_host"` `category:"advanced"`
	MaxConnsPerHost       int           `yaml:"max_conns_per_host"`  `category:"advanced"`
}
```

Reply:
I've been thinking about it and decided to make it symmetric with the S3 config

Reply:
Adding it in #4007